core/src/utils/forms/form-controller.ts

```diff
@@ -4,7 +4,7 @@
  * @internal
  * @prop el: The Ionic form component to reference
  */
-type AllowedFormElements = HTMLIonInputElement | HTMLIonToggleElement;
+type AllowedFormElements = HTMLIonCheckboxElement | HTMLIonInputElement | HTMLIonToggleElement;
```

Comment:
You may want to take what I have done here: https://github.com/ionic-team/ionic-framework/blob/FW-2764-placement/core/src/utils/forms/form-controller.ts#L1 so we don't need to keep adding additional elements to the type here.